cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/StreamSession.java

```diff
@@ -139,7 +141,16 @@ public Future<StreamResult> finalizeStreamAsync() throws IOException
                                  "SSTable range %s should be enclosed in the partition range %s",
                                  sstableWriter.getTokenRange(), tokenRange);
         // close the writer before finalizing stream
-        sstableWriter.close(writerContext);
+        try
+        {
+            sstableWriter.close(writerContext);
+        }
+        catch (IOException ioe)
+        {
+            // clean up the sstables and rethrow on I/O errors when closing
+            cleanupSSTables(LOGGER);
+            throw ioe;
+        }
```

Comment:
@lukasz-antoniak , the clean up is added here. When close is called and failed, all `prepareSStablesToSend` scheduled in the future will be no-op and it should be safe to remove all sstables eagerly.